### ISSUE TITLE
Implement Learning Rate Adjustment with ReduceLROnPlateau for Generators G1 and G2 Training

### ISSUE THREAD

**Issue by Othman-vram** [Othman-vram/edge-connect#14]
Currently, Generator `G1` and Generator `G2` are trained separately using Canny edges with a learning rate of `lr = 10e-4` until the loss plateaus. We need to add the learning rate adjustment step using PyTorch's `ReduceLROnPlateau` scheduler. Once the loss plateaus, we will reduce the learning rate and continue training until convergence.
